Docs: Templating notifications (#60109)
grafana/grafana: docs/sources/alerting/manage-notifications/template-notifications/template-examples.md

---
aliases:
keywords:
  - grafana
  - alerting
  - notifications
  - templates
  - example templates
title: Template examples
weight: 500
---

# Template examples

> Here are number of more complex examples that you might want to use as a reference for customizing your notifications.

Reviewer: ```suggestion
Here are number of more complex examples to use as a reference for customizing your notification messages.
```

```
There are 2 firing alert(s), and 1 resolved alert(s)

Firing alerts:

- alertname=Test 1 grafana_folder=GrafanaCloud has value(s) B=1
- alertname=Test 2 grafana_folder=GrafanaCloud has value(s) B=2

Resolved alerts:

- alertname=Test 3 grafana_folder=GrafanaCloud has value(s) B=0
```

The message is created from two templates: `email.message_alert` and `email.message`. The `email.message_alert` template is used to print the labels and values for each firing and resolved alert while the `email.message` template contains the structure of the email.

```
{{- define "email.message_alert" -}}
{{- range .Labels.SortedPairs }}{{ .Name }}={{ .Value }} {{ end }} has value(s)
{{- range $k, $v := .Values }} {{ $k }}={{ $v }}{{ end }}
{{- end -}}

{{ define "email.message" }}
There are {{ len .Alerts.Firing }} firing alert(s), and {{ len .Alerts.Resolved }} resolved alert(s)

{{ if .Alerts.Firing -}}
Firing alerts:
{{- range .Alerts.Firing }}
- {{ template "email.message_alert" . }}
{{- end }}
{{- end }}

{{ if .Alerts.Resolved -}}
Resolved alerts:
{{- range .Alerts.Resolved }}
- {{ template "email.message_alert" . }}
{{- end }}
{{- end }}

{{ end }}
```

> ## Templating the labels, annotations, SilenceURL and DashboardURL of all alerts

Reviewer: ```suggestion
## Template the labels, annotations, SilenceURL and DashboardURL of all alerts
```

Author: Should I remove the `the`?

```
Template labels, annotations, SilenceURL and DashboardURL of all alerts
```

Reviewer: yes
